internal/exec/exec_test.go

```diff
@@ -0,0 +1,64 @@
+package exec
+
+import (
+	"os"
+	"testing"
+	"unsafe"
+
+	"github.com/Microsoft/hcsshim/internal/winapi"
+	"golang.org/x/sys/windows"
+)
+
+func TestExecProcess(t *testing.T) {
+	var procThreadSize uintptr
+	if err := winapi.InitializeProcThreadAttributeList(nil, 1, 0, &procThreadSize); err != nil && err != windows.E_NOT_SUFFICIENT_BUFFER {
+		t.Fatalf("failed to call InitializeProcThreadAttributeList: %s", err)
+	}
+
+	procHeap, err := winapi.GetProcessHeap()
+	if err != nil {
+		t.Fatalf("GetProcessHeap failed: %s", err)
+	}
+
+	attrList, err := winapi.HeapAlloc(procHeap, 0, procThreadSize)
+	if err != nil {
+		t.Fatalf("HeapAlloc failed: %v\n", err)
+	}
+	defer winapi.HeapFree(procHeap, 0, attrList)
+
+	var procAttrList *winapi.PROC_THREAD_ATTRIBUTE_LIST
+	procAttrList = (*winapi.PROC_THREAD_ATTRIBUTE_LIST)(unsafe.Pointer(attrList))
+	if err = winapi.InitializeProcThreadAttributeList(procAttrList, 1, 0, &procThreadSize); err != nil {
+		t.Fatalf("InitializeProcThreadAttributeList - second call failed: %v\n", err)
+	}
+
+	// This is useless but it tests all the win32 calls working. A better use for this test would be creating
+	// this process to be a child of a different process other than the one launching it but we can't gurantee
+	// any other processes running/get their pids.
```

Comment:
Why?

Reply:
This is just one of the ways to test UpdateProcThreadAttribute. The PROC_THREAD_ATTRIBUTE_PARENT makes it so that you can launch a process as the child of whatever pid you pass in instead of it always being a child of the process who launched it. It's "useless" in this case as we're just passing in the pid of the process who's going to launch it anyways just to test that the winapi bindings all work. I have another test I added (not pushed yet obviously) that is more useful. It is the adding a process to a job object before it's started case.

Reply:
Does this make sense or would you want it elaborated on?

Reply:
Maybe instead you could say like:
```
The call to UpdateProcThreadAttribute here does nothing since the parent process pid is not changed, but this tests that all necessary win32 calls work in the end to end experience.  
```

Reply:
That sounds good to me. Updating